### PR TITLE
Allow to configure cgroupsv1 per nodepool

### PR DESCRIPTION
**Pull request by fiunchinho**
### What does this PR do?

Backport of https://github.com/giantswarm/cluster/pull/348

### CODE REVIEW

**Comment by glitchcrab** [comment]
please ignore me, wrong browser tab :man_facepalming:

**Comment by Gacko** [comment]
I changed the PR's title as there is, IMHO, no need to prefix it this way. Apart from that: LGTM! I guess you only cherry-picked, right?

I guess it's currently hard to impossible to test stuff like this by just running cluster test suites, so we might need to come up with a test release that uses the test version of `cluster-aws` that includes this version of `cluster` here.

File: helm/cluster/templates/containerd.yaml [hunk 1]
```diff
@@ -1,8 +1,20 @@
-{{- if or $.Values.providerIntegration.resourcesApi.controlPlaneResourceEnabled $.Values.providerIntegration.resourcesApi.machinePoolResourcesEnabled }}
+{{- if $.Values.providerIntegration.resourcesApi.machinePoolResourcesEnabled }}
+{{- range $nodePoolName, $nodePoolConfig := $.Values.global.nodePools | default $.Values.providerIntegration.workers.defaultNodePools }}
+{{- $_ := set $ "nodePool" (dict "name" $nodePoolName "config" $nodePoolConfig) }}
 apiVersion: v1
 kind: Secret
 metadata:
-  name: {{ include "cluster.resource.name" $ }}-containerd-{{ include "cluster.data.hash" (dict "data" (tpl ($.Files.Get "files/etc/containerd/config.toml") $) "salt" $.Values.providerIntegration.hashSalt) }}
+  name: {{ include "cluster.resource.name" $ }}-{{ $nodePoolName }}-containerd-{{ include "cluster.data.hash" (dict "data" (tpl ($.Files.Get "files/etc/containerd/workers-config.toml") $) "salt" $.Values.providerIntegration.hashSalt) }}
```
Comment (AndiDog) [inline helm/cluster/templates/containerd.yaml:7]: Let's hope nobody calls their node pool `controlplane` 😆

**Review by AndiDog — APPROVED** [review]

File: helm/cluster/templates/containerd.yaml (hunk 1 above)
Comment (fiunchinho) [reply helm/cluster/templates/containerd.yaml:7]: lol I didn't think of that :smile: I don't think it will happen though (infamous last words)

**Comment by fiunchinho** [comment]
I migrated a test cluster that was using cgroupsv1 in one of the node pools, and the configuration was migrated successfully.

**Pull request merged**